core/src/main/java/org/apache/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -394,4 +443,48 @@ private static MetadataObject.Type getType(String type) {
   private static String getEntityType(SecurableObject securableObject) {
     return securableObject.type().name();
   }
+
+  public static Map<Long, String> getMetadataObjectFullNames(Long metalakeId, List<Long> ids) {
+    Map<Long, String> catalogIdAndNameMap = getCatalogIdAndNameMap(metalakeId);
+    Map<Long, Map<Long, String>> schemaIdAndNameMap = getSchemaIdAndNameMap(metalakeId);
```

Comment:
We don't need to get all the schema of the metalake. We can get the schema id list according to the fileset list.